src/compiler.ts

```diff
@@ -192,11 +220,23 @@ export class Compiler implements Emitter {
    * @param files the files that were specified as input in the CLI invocation.
    */
   private _prepareForBuild(...files: string[]) {
-    this.buildTypeScriptConfig();
-    this.writeTypeScriptConfig();
+    if (this.userProvidedTypeScriptConfig) {
+      this.loadTypeScriptConfig();
+      this.validateTypeScriptConfig();
+    } else {
+      this.buildTypeScriptConfig();
+      this.validateTypeScriptConfig();
```

Comment:
Do we need to validate the one that we ourselves built?

Reply:
We don't have to, but I think it will be good to ensure that the generated config and the rules we have for it actually match. Could probably be done in a test though.